valet-key/ValetKey.Web/Controllers/ValuesController.cs

```diff
@@ -45,43 +48,38 @@ public StorageEntitySas Get()
                     });
             }
         }
-
+      
         /// <summary>
         /// We return a limited access key that allows the caller to upload a file to this specific destination for defined period of time
         /// </summary>
         private StorageEntitySas GetSharedAccessReferenceForUpload(string blobName)
-        {
-            var blobClient = this.account.CreateCloudBlobClient();
-            var container = blobClient.GetContainerReference(this.blobContainer);
+        {          
+            var blob = blobServiceClient.GetBlobContainerClient(this.blobContainer).GetBlobClient(blobName);
 
-            var blob = container.GetBlockBlobReference(blobName);
-
-            var policy = new SharedAccessBlobPolicy
-            {
-                Permissions = SharedAccessBlobPermissions.Write,
+            var storageSharedKeyCredential = new StorageSharedKeyCredential(blobServiceClient.AccountName, ConfigurationManager.AppSettings["AzureStorageEmulatorAccountKey"]);
 
-                // Create a signature for 5 min earlier to leave room for clock skew
-                SharedAccessStartTime = DateTime.UtcNow.AddMinutes(-5),
+            var blobSasBuilder = new BlobSasBuilder
 
-                // Create the signature for as long as necessary -  we can 
-                SharedAccessExpiryTime = DateTime.UtcNow.AddMinutes(5)
+            {
+                BlobContainerName = this.blobContainer,
+                BlobName = blobName,
+                Resource = "b",
+                StartsOn = DateTimeOffset.UtcNow,
+                ExpiresOn = DateTimeOffset.UtcNow.AddHours(1),
```

Comment:
NIT: The expires on used to be 5 mins in the future and now it's an hour.  Was that intentional?